tap_workday/streams/abstracts.py

```diff
@@ -331,3 +337,43 @@ def get_bookmark(self, state: Dict, stream: str, key: Any = None) -> int:
         if self.bookmark_value is None:
             self.bookmark_value = super().get_bookmark(state, stream)
         return self.bookmark_value
+
+
+class WorkdayTableStream(FullTableStream):
+    """Base for simple Workday SOAP streams.
+
+    Child classes should set:
+    - service_name: Workday service string (e.g., "Human_Resources")
+    - operation_name: SOAP operation to call (e.g., "Get_Organizations")
+    - data_key: leaf key inside Response_Data (e.g., "Organization")
+    - version (optional): default "v44.2"
+    """
+
+    service_name: str = ""
+    operation_name: str = ""
+    data_key: str = ""
+    version: str = "v44.2"
+
+    def get_client(self):
+        """Client for WorkdayTableStream."""
+        cfg = self.client.config
+        # Allow per-service override via config, e.g. human_resources_version
+        override_key = f"{self.service_name.lower()}_version"
+        version = cfg.get(override_key, cfg.get("version", self.version))
+        return Client(cfg, service=self.service_name, version=version)
+
+    def sync(self, state, transformer, parent_obj=None):
+        """Synchronize records for WorkdayTableStream using centralized client, supporting incremental syncs."""
+        client = self.get_client()
+        # Try to get bookmark/state for incremental syncs
+        updated_since = None
+        if hasattr(self, "get_bookmark"):
+            # Use the same logic as IncrementalStream
+            try:
+                updated_since = self.get_bookmark(state, self.tap_stream_id)
+            except Exception:
+                updated_since = None
```

Comment:
why do we need this in the FullTableStream, do we support querying by updated_since for fulltable?